tests/pr_merge_test.rs

```diff
@@ -1,23 +1,64 @@
 mod git_helper;
 mod mocks;
 
+use std::sync::Arc;
+
+use tempdir::TempDir;
+
+use octobot::config::Config;
+use octobot::db::Database;
 use git_helper::temp_git::TempGit;
 use mocks::mock_github::MockGithub;
 use octobot::github;
 use octobot::pr_merge;
+use octobot::repos;
+use octobot::slack::{self, SlackAttachmentBuilder};
+
+use failure::format_err;
+
+use mocks::mock_slack::MockSlack;
+
+struct PRMergeTest {
+    git: TempGit,
+    github: MockGithub,
+    config: Arc<Config>,
+    slack: MockSlack,
+}
+
+fn new_test() -> (PRMergeTest, TempDir) {
+    let temp_dir = TempDir::new("repos.rs").unwrap();
+    let db_file = temp_dir.path().join("db.sqlite3");
+    let db = Database::new(&db_file.to_string_lossy()).expect("create temp database");
+
+    let config = Arc::new(Config::new(db));
+    config.users_write().insert("the-pr-owner", "the.pr.owner").unwrap();
+    config
+        .repos_write()
+        .insert_info(&repos::RepoInfo::new("the-owner/the-repo", "the-review-channel")
+            .with_jira("SER").with_jira("CLI")
+            .with_force_push(true))
+        .expect("Failed to add some-user/some-repo");
+
+    // todo: configure channels/users
```

Comment:
imo probably don't need to configure / test this. choosing the right channel, etc. is tested elsewhere.